src/Microsoft.Azure.SignalR/HubHost/ServiceConnectionContext.cs

```diff
@@ -128,5 +130,33 @@ private static bool IsAuthenticatedUser(Claim[] claims)
 
             return false;
         }
+
+        private FeatureCollection BuildFeatures()
+        {
+            var features = new FeatureCollection();
+            features.Set<IConnectionHeartbeatFeature>(this);
+            features.Set<IConnectionUserFeature>(this);
+            features.Set<IConnectionItemsFeature>(this);
+            features.Set<IConnectionIdFeature>(this);
+            features.Set<IConnectionTransportFeature>(this);
+            features.Set<IHttpContextFeature>(this);
+            return features;
+        }
+
+        private HttpContext BuildHttpContext(OpenConnectionMessage message)
+        {
+            var httpContextFeatures = new FeatureCollection();
+            httpContextFeatures.Set<IHttpRequestFeature>(new HttpRequestFeature
+            {
+                Headers = new HeaderDictionary((Dictionary<string, StringValues>) message.Headers),
```

Comment:
Why do you have to cast here?

Reply:
Because `HeaderDictionary` only accepts a `Dictionary<string, StringValues>`.
https://docs.microsoft.com/en-us/dotnet/api/microsoft.aspnetcore.http.headerdictionary?view=aspnetcore-2.0#constructors